### ISSUE TITLE
Publish crates to `crates.io`

### ISSUE THREAD

**Issue by shazow** [foundry-rs/foundry#3895]
### Component

Forge, Cast, Anvil

### Describe the feature you would like

Foundry needs to start tagging versioned releases, and providing permanent release assets.

Ideally a v1.0 version would be great.

Until then, even a v0.x release set would be a useful starting point, especially for third-party packaging of foundry.

### Additional context

Previous: #3881

This issue is to allow people to track and subscribe to this topic.

**Comment by rkrasiuk** [comment]
@gakonst thoughts?

**Comment by gakonst** [comment]
Yeah we've already announced 0.2 but didn't tag it proper I think. Let's do a v0.3 release and have persistent binaries for it as well as allow them to be downloaded in the action.

I think we'll need to finally figure out our release culture / cadence and changelogs...

**Comment by gakonst** [comment]
(cc @onbjerg this might be a small one given you easily did the reth formula)

**Comment by onbjerg** [comment]
@gakonst we already have a foundry formula it's just borked because we don't have actual versions yet

**Comment by RensR** [comment]
Just having a more recent version that isn't pruned would help. We ran nightly for a long time but that changes things too often without warning so we had to fall back to a stable version. The latest non-pruned version is [Nightly (2023-08-02)](https://github.com/foundry-rs/foundry/releases/tag/nightly-ca67d15f4abd46394b324c50e21e66f306a1162d), which is almost 2 months behind. Can we get a new version that won't be pruned? Ideally with a some tag indicating that it will remain available and some schedule when new non-pruning versions will be released.

**Comment by RensR** [comment]
@gakonst is it by design that the unofficial long term supported released (the ones on the 2nd of each month that stay up) change hashes randomly? Not sure how or why, but we relied on the Oct 2nd release, which is still up, with hash `e0722a10b45859892ec3b998df958a9edc77c202`. Not when I go to that release it is the same but targets hash`5be158ba6dc7c798a6f032026fe60fc01686b33b`.

 Is there any way, before the proper versioning starts, to use Foundry with static versions?

**Comment by excalq** [comment]
Interested to hear when this may happen, as not having versioned or immutable tags makes team-wide devops quite challenging. Specifically when encountering breaking changes when each developer has run `foundry up` causing them to have different versions. This also makes CI/CD brittle. It would be a big improvement to have `v0.x` semantic versioning over the current mutable nightly tagging.

**Comment by RensR** [comment]
@excalq FYI we've solved it with a makefile that installs a specific version. Devs never run foundryup, and CI is locked to the same version as the makefile. Not perfect but it works pretty well.

**Comment by zerosnacks** [comment]
Related: https://github.com/foundry-rs/foundry/issues/8659, to make verify crate available as standalone lib; publish to crates.io

**Comment by grandizzy** [comment]
v0.3.0 is now pinned and permanent, working towards a v1.0

**Comment by sambacha** [comment]
releases can be published to the package registry as well, using the container registry you can post arbitrary binary blobs. homebrew does this: https://github.com/Homebrew/brew/blob/b753315b0b1e78b361612bf4985502bf9dca5582/Library/Homebrew/github_packages.rb#L196-L428

**Comment by MicaiahReid** [comment]
Do we think the [foundry-config](https://crates.io/crates/foundry-config/0.2.0) crate will be updated as part of this endeavor? Looks like it hasn't been updated for over a year.

**Comment by DaniPopes** [comment]
We've discussed this internally and concluded that releasing on crates.io is more trouble than it's worth. Reth is doing fine with users importing and extending it via Git dependencies.

**Comment by shazow** [comment]
SGTM. The original issue was about stable tagged releases, and we have that now, so I'm happy. :)

**Comment by citizen-stig** [comment]
Hey @DaniPopes 

I understand that publishing creates extra overhead, but relying on git dependencies prevents downstream users from publishing their own crates without vendoring. That significantly limits the broader adoption of reth as a library ecosystem.

Would it be possible to publish at least a subset of the crates on crates.io?

- reth-errors
- reth-primitives
- reth-primitives-traits
- reth-ethereum-primitives

Could you reconsider this? 
I’m also happy to help prepare them for release if that would make the process easier.
